keps/sig-auth/1205-bound-service-account-tokens/README.md

```diff
@@ -265,53 +258,161 @@ service account token on behalf of pods running on that node. The
- 
-This functionality can be implemented entirely in the kube-apiserver and does not 
-require cooperation by the kubelet beyond the standard functionality implemented 
-as part of [Service Account Token Volumes](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/svcacct-token-volume-source.md).
+1.  Compare the `kubernetes.io/warnafter` field with current time. If current
+    time is after `kubernetes.io/warnafter` field, it implies calling client is
+    not reloading token regularly.
+1.  Expose metrics to monitor number of legacy and stale token used.
+1.  Add annotation to audit events for legacy and stale tokens including
+    necessary information to locate problematic client.
 
-We will direct cluster administrators to adopt bound service account tokens and 
-turn on the command line flag to enable extended token. Fix clients if monitoring or 
-audit report that stale or legacy token are in use. The operator can turn off 
-the command line flag after no stale token is being used.
+### Graduation Criteria
 
+#### Alpha->Beta
 
+All known migration frictions have been fixed.
 
-### Graduation Criteria
+- PodSecurityPolicies that allow secrets but not projected volumes will
+  handicap the provision of token volumes.
+  - Fixed in https://github.com/kubernetes/kubernetes/pull/92006
+- In-cluster clients that don’t reload service account tokens will start
+  failing an hour after deployment.
+  - Fixed in https://github.com/kubernetes/kubernetes/issues/68164
+- Pods running as non root may not access the service account token.
+  - Fixed in https://github.com/kubernetes/kubernetes/pull/89193
 
 #### Beta -> GA Graduation
 
-- TBD
+New `ServiceAccount` admission controller WAI in Beta for >= 1 minor without
+significant issues.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+- **How can this feature be enabled / disabled in a live cluster?**
+
+  - Feature gate name: `BoundServiceAccountVolume`
+  - Components depending on the feature gate: kubelet, kube-apiserver
+  - Will enabling / disabling the feature require downtime of the control
+    plane? yes, need to restart kube apiserver.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? yes, need to restart kubelet.
+
+- **Does enabling the feature change any default behavior?** yes, pods'
+  service account tokens will not be long-lived and are not stored as Secrets
+  any more.
+
+- **Can the feature be disabled once it has been enabled (i.e. can we roll
+  back the enablement)?** yes.
+
+- **What happens if we reenable the feature if it was previously rolled
+  back?** the same as the first enablement.
+
+- **Are there any tests for feature enablement/disablement?** yes, unit tests
+  will cover this.
+
+### Scalability
+
+- **Will enabling / using this feature result in any new API calls?**
+
+  - API call type: `TokenRequest`
+  - estimated throughput: 1/pod.
```

Comment:
```suggestion
  - estimated throughput: 1/pod every 8 minutes.
```

Reply:
to be accurate, would be 1/(0.8*ExpirationSeconds) as kubelet will cache the results.

Reply:
what is the default expiration time for the admission-injected volumes?

Reply:
3607 seconds by default. the minimum is 10 minutes and that's why client-go and other clients will reload every 8 minutes.

Reply:
ah, ok... adjust to "1/pod every ~48 minutes"

Reply:
done